core/http/client/src/main/java/org/eclipse/rdf4j/http/client/BackgroundGraphResult.java

```diff
@@ -87,19 +87,25 @@ public void remove()
 	protected void handleClose()
 		throws QueryEvaluationException
 	{
-		try {
-			super.handleClose();
-		}
-		finally {
-			closed = true;
-			try {
-				in.close();
-			}
-			catch (IOException e) {
-				throw new QueryEvaluationException(e);
-			}
-			finally {
-				queue.close();
+		if (!closed) {
+			synchronized (this) {
+				if (!closed) {
+					closed = true;
+					try {
+						try {
+							super.handleClose();
+						}
+						finally {
+							in.close();
+						}
+					}
+					catch (IOException e) {
+						throw new QueryEvaluationException(e);
+					}
+					finally {
+						queue.close();
```

Comment:
Is there an historical reason behind queue.close being called here but not in BackgroundTupleResult?